cairo/src/utils/dict.cairo

```diff
@@ -41,3 +47,41 @@ func dict_squash{range_check_ptr}(
     %}
     return (squashed_dict_start=squashed_dict_start, squashed_dict_end=squashed_dict_end);
 }
+
+// A wrapper around dict_read that hashes the key before accessing the dictionary if the key
+// does not fit in a felt.
+// @param byte_length: The number of bytes in the key.
+// @param key_len: The number of felt values used to represent the key.
+// @param key: The key to access the dictionary.
+func hashdict_read_bytes{poseidon_ptr: PoseidonBuiltin*, dict_ptr: DictAccess*}(
```

Comment:
```suggestion
func hashdict_read{poseidon_ptr: PoseidonBuiltin*, dict_ptr: DictAccess*}(
```

the function becoming generic, I I think that `hashdict_read`, as opposed to `dict_read` is enough (may not be bytes as input)